### PR TITLE
Improve new comer docs

### PR DESCRIPTION
**Pull request by leplatrem**
* [x] Reformulate parts of quickstart
* [x] Add illustrations

### CODE REVIEW

**Comment by leplatrem** [comment]
@ametaireau @Natim @hiromipaw  review?

File: docs/rationale.rst [hunk 1]
```diff
@@ -24,14 +24,18 @@ expose a well defined HTTP protocol for:
...
 
 .. _Mozilla Services: https://wiki.mozilla.org/CloudServices
+.. _elsewhere: http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api
+
+The goal is to produce standardized APIs, which follow some
+well known patterns, encouraging genericity in clients code.
+
+Of course, *Cliquet* can be :term:`extended <extensible>` and customized in
+many ways. It can also be used in any kind of project, for its tooling, utilities
+and helpers.
```
Comment (hiromipaw) [inline docs/rationale.rst]: I am wondering if it is worth mentioning also http://jsonapi.org/ in the well known patterns part?

File: docs/rationale.rst [hunk 2]
```diff
@@ -80,9 +87,22 @@ Dependencies
...
+
+* *PostgreSQL* for storage;
+* *Redis* for key-value cache with expiration;
+* *StatsD* metrics;
+* *Sentry* reporting via logging;
+* *NewRelic* database profiling (*development*);
+* *Werkzeug* Python code profiling (*development*).
+
+Currently, the default authentication relies on :term:`Firefox Accounts`, but any
 :ref:`authentication backend supported by Pyramid can be used <configuration-authentication>`.
```
Comment (hiromipaw) [inline docs/rationale.rst]: :+1:

**Comment by hiromipaw** [comment]
Also thanks for letting me taking part in this review :smiley_cat:

File: docs/contributing.rst [hunk 3]
```diff
@@ -99,4 +99,4 @@ Definition of done
 IRC channel
 ===========
 
-Join ``#storage`` on ``irc.mozilla.org``!
+Join ``#storage`` on ``irc.mozilla.org`` !
```
Comment (ametaireau) [inline docs/contributing.rst]: Actually you don't need a space here :)

File: docs/installation.rst [hunk 4]
```diff
@@ -4,13 +4,11 @@ Installation
...
-By default, *Cliquet* persists the records and cache in `Redis <http://redis.io/>`_.
+By default, a *Cliquet* application persists the records and cache in a local
+ `Redis <http://redis.io/>`_.
 
-In-memory or `PostgreSQL <http://postgresql.org/>`_ storage backend can be enabled in
-:ref:`storage configuration <configuration-storage>`.
-
-See dedicated paragraphs below for more details about installation of these
-services.
+Later, using the :ref:`application configuration <configuration-storage>`,
```
Comment (ametaireau) [inline docs/installation.rst]: I would remove "later".

File: docs/quickstart.rst [hunk 5]
```diff
@@ -13,7 +13,7 @@ create a minimal application, or use its scaffolding tool:
 
 ::
 
-    pcreate -s starter MyProject
+    $ pcreate -s starter MyProject
```
Comment (ametaireau) [inline docs/quickstart.rst]: I always wondered if we need to add these `$` signs everywhere. They just tend to make copy/paste harder in my opinion.

File: docs/quickstart.rst [hunk 6]
```diff
@@ -41,41 +41,42 @@ just add some extra initialization code:
         return config.make_wsgi_app()
 
 
-.. autofunction:: cliquet.initialize
+By doing that, basic features like authentication, monitoring, error formatting,
+deprecation are now available.
```
Comment (ametaireau) [inline docs/quickstart.rst]: deprecation *indicators*

File: docs/rationale.rst (hunk 1 above)
Comment (ametaireau) [inline docs/rationale.rst]: See #254 for the discussion, we're still discussing this part.

File: docs/rationale.rst [hunk 7]
```diff
@@ -113,24 +135,39 @@ Context
...
+    :align: center
+
+An offline-first JavaScript library will be published [#]_, with the aim of providing
+some reusable code for any client that interacts with any *Cliquet*-based API.
+
+.. figure:: images/kinto-multi-client.png
+    :align: center
+
+In the long term, we envision a world where client and server applications are
+decorrelated from their data [#]_!
```
Comment (ametaireau) [inline docs/rationale.rst]: from where their data lives?

File: docs/quickstart.rst (hunk 5 above)
Comment (leplatrem) [inline docs/quickstart.rst]: I thought so, but I realize they give the context implicity. One knows that this is a command to run in a shell.

We can remove them if we don't agree :)

File: docs/rationale.rst [hunk 8]
```diff
@@ -113,24 +135,39 @@ Context
...
+.. figure:: images/self-hostable.png
+    :align: center
+
+Since the protocol is language independant and follows good HTTP/REST principles,
+in the long term *Cliquet* should become only one among several server implementations.
+
+.. note::
+
+    We encourage you to implement a clone of this project — using Node.js, Asyncio,
+    Go, Twisted or even Django — following :ref:`the same protocol <api-endpoints>`!
```
Comment (ametaireau) [inline docs/rationale.rst]: or anything else ;)

**Comment by ametaireau** [comment]
r+ with comments.

File: docs/quickstart.rst (hunk 5 above)
Comment (ametaireau) [inline docs/quickstart.rst]: We could add this information visually maybe? Oh well, that's not that important; we can keep it like that :)

**Pull request merged**